### ISSUE TITLE
Keyboard handler: Replace ev.keyCode with ev.key

### ISSUE THREAD

**Issue by IvanSanchez** [Leaflet/Leaflet#3367]
Reading #3362 reminded me of a problem I had just recently with keyboard events.

On most browsers, the keycodes for the arrow keys are 37-40. However, some versions of Safari broke havoc by changing those with 63232-5 (see http://www.quirksmode.org/js/keys.html ). I doubt panning the map with just the keyboard works consistently across Safari versions as of now.

The "proper way" of detecting keys is by using the keyboard events defined on DOM Events Level 3 (http://www.w3.org/TR/DOM-Level-3-Events/#events-keyboardevents).

In a nutshell: `if (ev.keyCode==37)` becomes `if (ev.key=='ArrowLeft')`; `if (ev.keyCode==187)` becomes `if (ev.key=='+')`, and so on.

Unfortunately MSIE8 doesn't support DOM Events Level 3, and some versions of Firefox report `ev.key='Left'` instead of `ev.key='ArrowLeft'`, so keycodes are still needed.

I don't know if this is worth implementing, or maybe we can just rely on adding new keycodes when stuff breaks up.

**Comment by mourner** [comment]
Hmm, it would be nice to know what exact Safari versions are affected and what keycodes they expect. DOM3 events is eventually the right way to go, but for now we can stay with keycodes if we cover most cases.

**Comment by IvanSanchez** [comment]
I think only Safari 3 is really affected, as keyboard behaviour was ""fixed"" circa 2007 as of Webkit rev525: https://www.mail-archive.com/webkit-dev@lists.webkit.org/msg01543.html

I swear I saw the 632xx keycodes in a modern safari a couple weeks ago, but cannot remember how to replicate the circumstances :-(

Anyway, it would be nice to deprecate DOM2 events eventually. Can we drop this one into the wishlist?

**Comment by mourner** [comment]
I think we don't need to worry about Safari 3, it's ancient and I can't imagine many people using it. Regarding DOM2, — currently IE7+ should still be supported (in near future may bump to IE8) so even if we go with DOM3, there should be a fallback.

**Comment by mourner** [comment]
Closing for now unless there's a real problem with this.
